serviceAccessConfig/lib/serviceAccessConfig/apache.py

```diff
@@ -0,0 +1,147 @@
+#  Copyright (C) 2016 SUSE LLC, Robert Schweikert <rjschwei@suse.com>
+#  All rights reserved.
+#
+#  This file is part of serviceAccessConfig
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+"""serviceAccessConfig plugin for Apache"""
+
+import ConfigParser
+import glob
+import logging
+import os
+import re
+
+from accessrulegenerator import ServiceAccessGenerator
+from generatorexceptions import *
+
+apache_directory_directive = re.compile(r'\s*<Directory .*>')
+
+
+class ServiceAccessGeneratorApache(ServiceAccessGenerator):
+    """Specific access rule generator for Apache Web Server"""
+
+    # ======================================================================
+    def __init__(self, ip_source_config_file_name):
+        self.section_name = 'apache'
+        super(ServiceAccessGeneratorApache, self).__init__(
+            ip_source_config_file_name)
+
+    # ======================================================================
+    def _get_apache_ip_directive(self):
+        """Figure out the version of Apache and return the appropriate
+           Apache directive for IP address filtering"""
+        apachectl = None
+        directive = 'Allow from'
+        if os.path.exists('/usr/sbin/apache2ctl'):
+            apachectl = '/usr/sbin/apache2ctl'
+        else:
+            apachectl = '/usr/sbin/apachectl'
+
+        if not os.access(apachectl, os.X_OK):
+            # Do some really ugly stuff
+            upgradeScript = glob.glob('/usr/share/apache2/*-upgrade')
+            for scr in upgradeScript:
+                toVer = scr.split('-')[2]
+                if int(toVer) >= 24:
+                    directive = 'Require ip'
+                    break
+            return directive
+
+        apacheInfo = os.popen('%s -V' % apachectl).readlines()
+        for ln in apacheInfo:
```

Comment:
I see no advantage for a regex at this point as I still have to pull out the the matches to form the int for version comparison.